src/components/RepoList/DeleteRepoDialog.tsx

```diff
@@ -0,0 +1,148 @@
+"use client"
+
+import { Button } from "@/components/ui/button"
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog"
+import {
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerDescription,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from "@/components/ui/drawer"
+import { useMediaQuery } from "@/hooks/useMediaQuery"
+import { TrashIcon } from "@radix-ui/react-icons"
+import * as React from "react"
+import { toast } from "sonner"
+import { Icons } from "../icons"
+import { handleDelete } from "./actions"
+import { useRouter } from "next/navigation"
+
+interface DeleteRepoDialogProps
+  extends React.ComponentPropsWithoutRef<typeof Dialog> {
+  repoId: string
+  showTrigger?: boolean
+  onSuccess?: () => void
+}
+
+export function DeleteRepoDialog({
+  showTrigger = true,
+  onSuccess,
+  repoId = "",
+  ...props
+}: DeleteRepoDialogProps) {
+  const router = useRouter()
+  const [isDeletePending, startDeleteTransition] = React.useTransition()
+  const isDesktop = useMediaQuery("(min-width: 640px)")
+
+  function onDelete() {
+    startDeleteTransition(async () => {
+      try {
+        const [res] = await handleDelete(repoId)
+        router.push("/dashboard")
+        onSuccess?.()
+        toast.success(
+          `Repository titled ${res.name} has been deleted successfully!`,
+        )
+      } catch (error: unknown) {
+        const message = error instanceof Error ? error.message : ""
+        toast.error(message)
```

Comment:
Shall we display a default message instead of an empty string?

Reply:
What are your thoughts on this?

Reply:
I don't think we're gonna localize it anytime soon and having all strings on a separate file doesn't add any value (adds unused strings actually, as we refactor often at this stage).

Here my point was only on that in these two line we might end up with `toast.error("")`. I think we should avoid showing an empty error toast and make sure there's always a message.